src/js/components/SpaceModal.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+import React, {useState} from "react"
+
+import ModalBox from "./ModalBox/ModalBox"
+import {useDispatch, useSelector} from "react-redux"
+import {reactElementProps} from "../test/integration"
+import Modal from "../state/Modal"
+import Spaces from "../state/Spaces"
+import {Input, InputSubmit} from "./form/Inputs"
+import Form from "./form/Form"
+import renameSpace from "../flows/renameSpace"
+
+export default function SpaceModal() {
+  return (
+    <ModalBox
+      name="space"
+      title=""
+      buttons={[]}
+      {...reactElementProps("spaceModal")}
+    >
+      <SpaceModalContents />
+    </ModalBox>
+  )
+}
+
+const SpaceModalContents = () => {
+  const {spaceId, clusterId} = useSelector(Modal.getArgs)
+  const space = useSelector(Spaces.get(clusterId, spaceId))
+  const {name} = space || ""
```

Comment:
An undefined space will cause an error here because it will try to destructure the empty string.

You might want `const {name} = space || {name: ""}`